packages/contracts/contracts/core/DAO.sol

```diff
@@ -286,6 +291,28 @@ contract DAO is
         emit StandardCallbackRegistered(_interfaceId, _callbackSelector, _magicNumber);
     }
 
+    /// @inheritdoc IDAO
+    function daoURI() external view override returns (string memory) {
+        return _daoURI;
+    }
+
+    /// @inheritdoc IDAO
+    function setDaoURI(string calldata newDaoURI)
+        external
+        override
+        auth(address(this), SET_METADATA_PERMISSION_ID)
+    {
+        _setDaoURI(newDaoURI);
+    }
+
+    /// @notice Sets the new DAO uri and emits the associated event.
+    /// @param newDaoURI The new DAO uri.
+    function _setDaoURI(string calldata newDaoURI) internal {
+        _daoURI = newDaoURI;
+
+        emit DaoURIUpdated(newDaoURI);
```

Comment:
in other words, I'm not sure about it. Needs research how etherscan detects the change of the URI.

Reply:
`emit NewDaoURI(...)`

Reply:
if you checkout the EIP they use `NewURI` when you use a registry for the URI handling. I suggest use the same event as mentioned there without the address, because they don't to it either